### PR TITLE
Theme JSON: add global block gap css vars

### PR DESCRIPTION
**Pull request by ramonjd**
Initial commit to test how and where to manipulate the style declarations for block gap.

See: https://github.com/WordPress/gutenberg/pull/35454

Trac ticket: 

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/class-wp-theme-json.php [hunk 1]
```diff
@@ -766,14 +776,46 @@ private function get_block_classes( $style_nodes ) {
...
 					$declarations_duotone[] = $declaration;
 				}
+				// 2. Save the block gap row and column CSS vars
+				if ( str_starts_with( $declaration['name'], '--wp--style--block' ) ) {
+					$declarations_block_gap[ $declaration['name'] ] = $declaration;
+					unset( $declarations[ $index ] );
+				}
+			}
+
+			if ( ! empty( $declarations_block_gap ) ) {
```
Comment (ramonjd) [inline src/wp-includes/class-wp-theme-json.php:789]: This is the first attempt, which will be the worst attempt. 

Just getting things to work as expected before refactoring.

**Comment by oandregal** [comment]
I commented at https://github.com/WordPress/gutenberg/pull/35454#issuecomment-990891974 asking whether this was for WordPress 5.9 or 6.0. If it's the later, we shouldn't port this just yet.

**Comment by ramonjd** [comment]
> I commented at WordPress/gutenberg#35454 (comment) asking whether this was for WordPress 5.9 or 6.0. If it's the later, we shouldn't port this just yet.

Thanks a lot @oandregal !!

I was just experimenting here, trying to find the best point which we could filter the block class declarations.

My hope was to only add something like this in [wp-theme-json](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/class-wp-theme-json.php#L778)

`apply_filters( 'gutenberg_block_classes_declarations', $declarations )`

and move the logic to the plugin.

I am in no way certain that's the right way forward yet so I would be very glad to hear your advice. 😄

**Comment by oandregal** [comment]
I don't think we want to add low-level filters at the moment. We discussed adding a couple of high-level ones at https://github.com/WordPress/gutenberg/pull/36909 but decided against it.

**Pull request closed without merging**